packages/editor/src/plugins/image-gallery/components/add-images.tsx

```diff
@@ -1,11 +1,49 @@
 import IconChoose from '@editor/editor-ui/assets/plugin-icons/image-gallery/icon-layered-images.svg'
+import { EditorPluginType } from '@editor/types/editor-plugin-type'
+import { v4 as uuidv4 } from 'uuid'
 
-export function AddImages() {
+import type { ImageGalleryProps } from '..'
+
+export function createImagePlugin(url: string, caption: string) {
+  return {
+    plugin: EditorPluginType.Image,
+    state: {
+      src: url,
+      caption: {
+        plugin: EditorPluginType.Text,
+        state: [
+          {
+            type: 'p',
+            children: [
+              {
+                text: caption,
+              },
+            ],
+          },
+        ],
+        id: uuidv4(),
+      },
+    },
+    id: uuidv4(),
+  }
+}
+
+interface AddImagesProps extends ImageGalleryProps {
+  onAddImages: () => void
+}
+
+export function AddImages(props: AddImagesProps) {
+  const { state, onAddImages } = props
+  const onClickAddImages = () => {
+    state.images.insert(0, createImagePlugin('', ''))
+    onAddImages()
+  }
```

Comment:
Is there a need for splitting this logic across two components? Maybe just:
```suggestion
  const { onAddImages } = props
```
(continued in next comments)

Imo, splitting the logic across files makes it harder to understand, plus you need to pass more stuff around.

Reply:
Done :heavy_check_mark: